packages/vulcan-core/lib/modules/default_mutations.js

```diff
@@ -82,164 +83,197 @@ export const getDefaultMutations = (collectionName, options = {}) => {
+    mutations.edit = updateMutation;
+  }
+  if (options.upsert) {
     // mutation for upserting a specific document
-    upsert: {
-      name: `${collectionName}Upsert`,
-
-      description: `Mutation for upserting a ${collectionName} document`,
+    mutations.upsert = {
+      description: `Mutation for upserting a ${typeName} document`,
 
-      async mutation(root, { search, set, unset }, context) {
-        const collection = context[collectionName];
+      async mutation(root, { input }, context) {
+        const { selector, data } = input;
+        const collection = context[typeName];
 
         // check if document exists already
-        const existingDocument = await Connectors.get(collection, search, { fields: { _id: 1 } });
+        const existingDocument = await Connectors.get(collection, selector, { fields: { _id: 1 } });
 
         if (existingDocument) {
-          const editArgs = {
-            documentId: existingDocument._id,
-            set,
-            unset,
-          };
-          return await collection.options.mutations.edit.mutation(root, editArgs, context);
+          return await collection.options.mutations.update.mutation(root, { selector, data }, context);
         } else {
-          return await collection.options.mutations.new.mutation(root, { document: set }, context);
+          return await collection.options.mutations.create.mutation(root, { data }, context);
         }
       },
-    },
-
+    };
+  }
+  if (options.delete) {
     // mutation for removing a specific document (same checks as edit mutation)
 
-    remove: {
+    const deleteMutation = {
+      description: `Mutation for deleting a ${typeName} document`,
 
-      name: `${collectionName}Remove`,
-      
-      description: `Mutation for deleting a ${collectionName} document`,
-      
       check(user, document) {
         if (options.removeCheck) {
           return options.removeCheck(user, document);
         }
-        
+
         if (!user || !document) return false;
-        return Users.owns(user, document) ? Users.canDo(user, `${collectionName.toLowerCase()}.remove.own`) : Users.canDo(user, `${collectionName.toLowerCase()}.remove.all`);
+        return Users.owns(user, document)
+          ? Users.canDo(user, `${typeName.toLowerCase()}.remove.own`)
+          : Users.canDo(user, `${typeName.toLowerCase()}.remove.all`);
       },
-      
-      async mutation(root, {documentId}, context) {
 
-        const collection = context[collectionName];
+      async mutation(root, { input }, context) {
+        const { selector } = input;
+        const { documentId } = selector;
+        const collection = context[typeName];
 
         const document = await Connectors.get(collection, documentId);
         Utils.performCheck(this.check, context.currentUser, document, context);
 
-        return await removeMutator({
-          collection, 
-          documentId: documentId, 
+        return await deleteMutator({
```

Comment:
Similar comment as above, we should be passing the entire selector to the `deleteMutator` and let it handle what to do with it.